nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -1247,13 +1280,85 @@ def dummy():
                 log_probs, token_ids = sample_token_fn(logits=output_tensor[:, -1, :])
                 # enforce valid range of token ids
                 token_ids = torch.clamp(token_ids, max=tokenizer.vocab_size - 1)
-                # collect all predicted tokens and log_probs
-                predicted_tokens_dec = torch.cat(
-                    [predicted_tokens_dec.to(token_ids.device), token_ids.unsqueeze(1)], dim=1
-                )
-                predicted_log_probs = torch.cat(
-                    [predicted_log_probs.to(log_probs.device), log_probs.unsqueeze(1)], dim=1
-                )
+
+                if i == 0 and beam_search:
```

Comment:
Let's move the code blocks into method in `nemo/collections/nlp/modules/common/text_generation_utils.py`